rasterio/env.py

```diff
@@ -430,6 +440,26 @@ def wrapper(*args, **kwds):
             session = DummySession()
 
         with env_ctor(session=session):
+
+            # We'll try to rely on system certs, but if curl finds
+            # none we will try certifi.
+            try:
+                return f(*args, **kwds)
+            except RasterioIOError as exc:
+                if str(exc).startswith("CURL error: error setting certificate"):
+                    try:
+                        import certifi
+                    except ImportError:
+                        raise CertificateError(
+                            "Set CURL_CA_BUNDLE to locate certificate or install certifi"
+                        )
+                    os.environ.setdefault("CURL_CA_BUNDLE", certifi.where())
+                else:
+                    raise
+
+            # The VSI cache will be contaminated by the previous
+            # failure and must be cleared before we re-try.
+            clear_vsi_curl_cache()
```

Comment:
@snowman2 since OS X and conda-forge builds don't need this, I've built a re-try with certifi into our wrapper around `rasterio.open()`.

Reply:
TODO: a test